R/tables.R

```diff
@@ -2,12 +2,13 @@
 MB = NCOL = NROW = NULL
 
 tables = function(mb=TRUE, order.col="NAME", width=80,
-                   env=parent.frame(), silent=FALSE, index=FALSE)
+                  env=parent.frame(), silent=FALSE, index=FALSE)
 {
   # Prints name, size and colnames of all data.tables in the calling environment by default
-  all_obj = objects(envir=env, all.names=TRUE)
-  is_DT = which(vapply_1b(all_obj, function(x) is.data.table(get(x, envir=env))))
-  if (!length(is_DT)) {
+  # include "hidden" objects (starting with .) via all.names=TRUE, but exclude ... specifically, #5197
+  all_obj = grep("...", objects(envir=env, all.names=TRUE, sorted=order.col == "NAME"), invert=TRUE, fixed=TRUE, value=TRUE)
+  is_DT = vapply_1b(mget(all_obj, envir=env), is.data.table)
```

Comment:
changing from `get` to `mget` here meant that `inherits` was changed from TRUE to FALSE because the defaults are different. But probably it should be FALSE anyway so seems like a good change in that respect.